tools/loadtesting/terraform/ecs.tf

```diff
@@ -0,0 +1,385 @@
+resource "aws_alb" "main" {
+  name            = "fleetdm"
+  internal        = false
+  security_groups = [aws_security_group.lb.id, aws_security_group.backend.id]
+  subnets         = module.vpc.public_subnets
+  idle_timeout    = 600
+}
+
+resource "aws_alb_target_group" "main" {
+  name                 = "fleetdm"
+  protocol             = "HTTP"
+  target_type          = "ip"
+  port                 = "8080"
+  vpc_id               = module.vpc.vpc_id
+  deregistration_delay = 30
+
+  load_balancing_algorithm_type = "least_outstanding_requests"
+
+  health_check {
+    path                = "/healthz"
+    matcher             = "200"
+    timeout             = 10
+    interval            = 15
+    healthy_threshold   = 5
+    unhealthy_threshold = 5
+  }
+
+  depends_on = [aws_alb.main]
+}
+
+resource "aws_alb_listener" "https-fleetdm" {
+  load_balancer_arn = aws_alb.main.arn
+  port              = 443
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-FS-1-2-Res-2019-08"
+  certificate_arn   = aws_acm_certificate_validation.dogfood_fleetdm_com.certificate_arn
+
+  default_action {
+    target_group_arn = aws_alb_target_group.main.arn
+    type             = "forward"
+  }
+}
+
+resource "aws_alb_listener" "http" {
+  load_balancer_arn = aws_alb.main.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_ecs_cluster" "fleet" {
+  name = "${local.prefix}-backend"
+
+  setting {
+    name  = "containerInsights"
+    value = "enabled"
+  }
+}
+
+resource "aws_ecs_service" "fleet" {
+  name                               = "fleet"
+  launch_type                        = "FARGATE"
+  cluster                            = aws_ecs_cluster.fleet.id
+  task_definition                    = aws_ecs_task_definition.backend.arn
+  desired_count                      = 10
+  deployment_minimum_healthy_percent = 100
+  deployment_maximum_percent         = 200
+  health_check_grace_period_seconds  = 30
+
+  load_balancer {
+    target_group_arn = aws_alb_target_group.main.arn
+    container_name   = "fleet"
+    container_port   = 8080
+  }
+
+  network_configuration {
+    subnets         = module.vpc.private_subnets
+    security_groups = [aws_security_group.backend.id]
+  }
+
+  depends_on = [aws_alb_listener.http, aws_alb_listener.https-fleetdm]
+}
+
+resource "aws_cloudwatch_log_group" "backend" {
+  name              = "fleetdm"
+  retention_in_days = 1
+}
+
+data "aws_region" "current" {}
+
+data "aws_secretsmanager_secret" "license" {
```

Comment:
for load testing purposes, I'd remove secrets manager related stuff. If we need license for EE related feature, just feed in via environment variable.

Deleting/recreating environments has turned out to be difficult with secrets manager usage and its default 30 day secret expiry window

Reply:
I solve that by using a pet name. So if it deletes/recreates it gets a new name.

Reply:
fair enough, I still think secrets manager offers very little for load-test environment.

Reply:
I agree, but it'll cause endless red flags from security minded people thinking they found something real.

Reply:
Since this is an open source project.